### PR TITLE
AirLoopHVACUnitarySystem FT fixes

### LINKED ISSUE #4715: Creating a VS WSHP using UnitarySystem object generates incorrect air flow ratios for UnitarySystemPerformance:Multispeed object
Created a Variable Speed WSHP programmatically using *OS:AirLoopHVAC:UnitarySystem* object in Openstudio V3.4 but the *ForwardTranslateAirLoopHVACUnitarySystem* code fails to generate the *UnitarySystemPerformance:Multispeed* object 
correctly. See sample energyplus object output created for water-source VS coils:

```
UnitarySystemPerformance:Multispeed,
  SZ-SYS SPACE1-1 Thermal Zone Unitary System Performance, !- Name
  5,                                      !- Number of Speeds for Heating
  5,                                      !- Number of Speeds for Cooling
  No,                                     !- Single Mode Operation
  ,                                       !- No Load Supply Air Flow Rate Ratio
  1,                                      !- Heating Speed Supply Air Flow Ratio 1
  1,                                      !- Cooling Speed Supply Air Flow Ratio 1
  1,                                      !- Heating Speed Supply Air Flow Ratio 2
  1,                                      !- Cooling Speed Supply Air Flow Ratio 2
  ,                                       !- Heating Speed Supply Air Flow Ratio 3
  ,                                       !- Cooling Speed Supply Air Flow Ratio 3
  ,                                       !- Heating Speed Supply Air Flow Ratio 4
  ,                                       !- Cooling Speed Supply Air Flow Ratio 4
  ,                                       !- Heating Speed Supply Air Flow Ratio 5
  ;                                       !- Cooling Speed Supply Air Flow Ratio 5
```

Inspecting the ForwardTranslateAirLoopHVACUnitarySystem code I'm suspicious of this section https://github.com/NREL/OpenStudio/blob/6a935cab2ae26b4170caf107287df201f4612401/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp#L573-L610

Also, I see a typo [CoilCoolingDXMultiSpeed](https://github.com/NREL/OpenStudio/blob/develop/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp#L481). This DX cooling coil type should have been *CoilCoolingDXVariableSpeed*, I think. I ran into the same problem when I tried to created a VS air-source heat pump using *OS:AirLoopHVAC:UnitarySystem* as a parent. 

https://github.com/NREL/OpenStudio/blob/6a935cab2ae26b4170caf107287df201f4612401/src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp#L480-L482
 
## Environment ##
Some additional details about your environment for this issue (if relevant):
-    Platform (Operating system, version): Windows
-    Version of OpenStudio (if using an intermediate build, include SHA): 3.4.0
-    Defect file attached

[in.osm.txt](https://github.com/NREL/OpenStudio/files/9788780/in.osm.txt)

### PR DESCRIPTION
**Pull request by joseph-robertson**
Pull request overview
---------------------

 - Fixes #4715

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp [hunk 1]
```diff
@@ -591,6 +606,14 @@ namespace energyplus {
           } else {
             extensible.setString(UnitarySystemPerformance_MultispeedExtensibleFields::HeatingSpeedSupplyAirFlowRatio, "Autosize");
           }
+        } else if (static_cast<unsigned>(i) < waterToAirHeatingStages.size()) {
```
Comment (joseph-robertson) [inline src/energyplus/ForwardTranslator/ForwardTranslateAirLoopHVACUnitarySystem.cpp:609]: Add support for missing `CoilHeatingWaterToAirHeatPumpVariableSpeedEquationFitSpeedData`.

**Review by joseph-robertson — COMMENTED** [review]
There are certainly some obvious bugs in the FT, but there are other issues that are less obvious whether they are intentional or not. Definitely need to run this through OS-resources.

**Comment by jmarrec** [comment]
> There are certainly some obvious bugs in the FT, but there are other issues that are less obvious whether they are intentional or not. Definitely need to run this through OS-resources.

have you ran the reg tests?

**Comment by joseph-robertson** [comment]
I have not.

**Pull request merged**